sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2739,6 +2752,26 @@ class Dataset[T] private[sql](
   }
 
   /**
+   * Returns true when the Dataset is cached and materialized.
+   *
+   * @group basic
+   * @since 2.3.0
+   */
+  def isMaterialized(): Boolean = {
```

Comment:
What is the use scenarios for this API? RDD also doesn't have the API to check it's materialized. I think in practice, we don't need to check if an Dataset/RDD is materialized, except for test purpose.